pkg/prometheus/promcfg.go

```diff
@@ -2577,6 +2586,10 @@ func (cg *ConfigGenerator) generateScrapeConfig(
 		cfg = cg.WithMinimumVersion("2.49.0").AppendMapItem(cfg, "scrape_protocols", sc.Spec.ScrapeProtocols)
 	}
 
+	if sc.Spec.JobName != nil || sc.Spec.RelabelConfigs != nil {
```

Comment:
not sure why this new block is needed?

Reply:
I must've overlooked it from the original commits.